src/panels/Menu/TaskBar/TaskBar.tsx

```diff
@@ -0,0 +1,40 @@
+import { Box, Flex } from '@mantine/core';
+import { WindowLayoutOptions } from 'src/windowmanagement/WindowLayout/WindowLayout';
+
+import { menuItemsData } from '@/windowmanagement/data/MenuItems';
+
+import { TaskBarMenuButton } from './TaskBarMenuButton';
+
+interface Props {
+  addWindow: (component: React.JSX.Element, options: WindowLayoutOptions) => void;
```

Comment:
use addwindow from the provider instead? visiblemenuitems could also be part of the provider